modules/integration/src/test/java/io/ballerina/messaging/broker/integration/standalone/jms/QueueLocalTransactionRollbackTest.java

```diff
@@ -111,40 +120,28 @@ public void testConsumerRollbackTransaction(String port) throws NamingException,
         for (int i = 0; i < numberOfMessages; i++) {
             producer.send(producerSession.createTextMessage("Test message " + i));
         }
-        //commit all sent messages
-        producerSession.commit();
-        producerSession.close();
+        // rollback all sent messages
+        producerSession.rollback();
 
-        // Consume published messages
+        // consume messages
         Session subscriberSession = connection.createSession(true, Session.SESSION_TRANSACTED);
         Destination subscriberDestination = (Destination) initialContextForQueue.lookup(queueName);
         MessageConsumer consumer = subscriberSession.createConsumer(subscriberDestination);
 
-        for (int i = 0; i < numberOfMessages; i++) {
-            Message message = consumer.receive(5000);
-            Assert.assertNotNull(message, "Message #" + i + " was not received");
-        }
-        //rollback all received messages
-        subscriberSession.rollback();
-
-        int numberOfMessagesAfterRollback = 0;
-        for (int i = 0; i < numberOfMessages; i++) {
-            Message message = consumer.receive(5000);
-            Assert.assertNotNull(message, "Message #" + i + " was not received after rollback");
-            numberOfMessagesAfterRollback++;
-        }
-        Assert.assertEquals(numberOfMessages, numberOfMessagesAfterRollback, "Only "
-                + numberOfMessagesAfterRollback + " messages received" + " after rollback but expect "
-                + numberOfMessages + " messages");
+        Message message = consumer.receive(1000);
+        Assert.assertNull(message, "Messages should not receive upon publisher rollback");
 
+        producerSession.close();
         subscriberSession.close();
         connection.close();
     }
 
     @Parameters({"broker-port"})
     @Test
-    public void testProducerRollbackTransaction(String port) throws NamingException, JMSException {
-        String queueName = "testProducerRollbackTransaction";
+    public void testTwoConsumersRollbackOneProducerCommitTransaction(String port)
+            throws NamingException, JMSException {
+        System.setProperty("STRICT_AMQP", "true");
```

Comment:
What if we use STRICT_AMQP for all the test cases. Maybe for all the transaction tests as a start?

Reply:
Fixed in 633fe86.